pytorch_lightning/core/hooks.py

```diff
@@ -204,17 +204,23 @@ def on_test_batch_end(self, outputs: Any, batch: Any, batch_idx: int, dataloader
         """
         # do something when the batch ends
 
+    def on_test_model_train(self) -> None:
+        """
+        Sets the model to train during the test loop
+        """
+        self.train()
```

Comment:
This is a confusing hook. When would that be called?

Reply:
At the end of evaluation loop to put back the model to train model. I agree, it is weird.